proposals/automated-go-ipfs-releases.md

```diff
@@ -0,0 +1,121 @@
+
+Initial PR: TBD <!-- Reference the PR first proposing this document. Oooh, self-reference! -->
+
+<!--
+This template is for a proposal/brief/pitch for a significant project to be undertaken by a Web3 Dev project team.
+The goal of project proposals is to help us decide which work to take on, which things are more valuable than other things.
+-->
+<!--
+A proposal should contain enough detail for others to understand how this project contributes to our team’s mission of product-market fit
+for our unified stack of protocols, what is included in scope of the project, where to get started if a project team were to take this on,
+and any other information relevant for prioritizing this project against others.
+It does not need to describe the work in much detail. Most technical design and planning would take place after a proposal is adopted.
+Good project scope aims for ~3-5 engineers for 1-3 months (though feel free to suggest larger-scoped projects anyway). 
+Projects do not include regular day-to-day maintenance and improvement work, e.g. on testing, tooling, validation, code clarity, refactors for future capability, etc.
+-->
+<!--
+For ease of discussion in PRs, consider breaking lines after every sentence or long phrase.
+-->
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve. Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives. 
+-->
+
+Currently the process of performing a go-ipfs release is tedious and is a barrier to having smaller more frequent releases. Having a more automated release process would allow us to iterate faster.
+
+Additionally, it would be great if our release process was such that we felt comfortable having an option for go-ipfs to automatically update on start instead of making users update. This would also be a step towards removing user barriers with package managers like `snap` which are not CLI tool friendly.
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+- Doing go-ipfs releases manually must impede release cycle time in a way that impacts project velocity, or
+- The way in which users currently update go-ipfs is sufficiently an impediment that either it causes users to have a bad experience with the stack (e.g. by running into bugs which have long been fixed), or it significantly harms development velocity by needing to take into account old non-upgraded nodes
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+A go-ipfs maintainer could run a script or GitHub action that would automatically perform a release from a given branch instead of manually performing the release.
```

Comment:
I'm wondering if this should expand.  I'm wondering something like the following as an example:

Once a commit is pushed to the release branch, in addition to our tests, it automatically builds/deploys to PL infrastructure in a controlled way (e.g., small region on day 1 (catch glaring bugs), big region on day 2 (catch issues that only show up under load), remaining regions on day 3).  If at any point our alarms trip, we have autorollback, the deployment pipeline blocks, and we get notified during business hours about the bug to fix.  

I think the value of this would be that we would have high confidence on whenever we want to cut/market a new release version as it has been baking on PL infrastructure in production already and if there is a production issue we can quickly pinpoint the change that caused it.